outlines/generate/generator.py

```diff
@@ -126,6 +127,29 @@ def token_generator(model, sampler: "Sampler") -> Callable:
 
     """
 
+    def cache_key_args_for_generate(
```

Comment:
Wouldn't we generate an unnecessarily large cache by caching the output of `generate_token` instead of the entire sequence?